Libs/MRML/Core/vtkMRMLTextNode.cxx

```diff
@@ -0,0 +1,148 @@
+/*=auto=========================================================================
+
+  Portions (c) Copyright 2009 Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+  See Doc/copyright/copyright.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Program:   3D Slicer
+  Module:    $RCSfile: vtkMRMLTextNode.cxx,v $
+  Date:      $Date: 2006/03/19 17:12:29 $
+  Version:   $Revision: 1.3 $
+
+=========================================================================auto=*/
+
+// SlicerOpenIGTLink MRML includes
+#include "vtkMRMLTextNode.h"
+#include "vtkMRMLTextStorageNode.h"
+
+// MRML includes
+#include <vtkMRMLScene.h>
+
+// VTK includes
+#include "vtkObjectFactory.h"
+#include "vtkXMLUtilities.h"
+
+const int LENGTH_BEFORE_STORAGE_NODE = 256;
+
+//----------------------------------------------------------------------------
+vtkMRMLNodeNewMacro(vtkMRMLTextNode);
+
+//-----------------------------------------------------------------------------
+vtkMRMLTextNode::vtkMRMLTextNode()
+  : Text("")
+  , Encoding(VTK_ENCODING_US_ASCII)
+  , ForceStorageNode(false)
```

Comment:
It would be better to use the variable name ForceCreateStorageNode with enum value: CreateStorageNodeAuto, CreateStorageNodeAlways, CreateStorageNodeNever (to make the behavior more predictable).